src/adaptors/giveth/index.js

```diff
@@ -0,0 +1,380 @@
+const urlGivEconomyGnosis =
+  'https://api.thegraph.com/subgraphs/name/giveth/giveth-economy-xdai';
+const urlBalancer =
+  'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2';
+const urlIchi = 'https://api.ichi.org/v1/farms/20009';
+
+// GIV Token
+const givTokenMainnetAddress = '0x900db999074d9277c5da2a43f252d74366230da0';
+const givTokenGnosisAddress = '0x4f4f9b8d5b4d0dc10506e5551b0513b61fd59e75';
+
+// GIV 100% UNIPOOL LM
+const givMainnetContractInfo = '0x4b9efae862a1755f7cecb021856d467e86976755';
+const givGnosisContractInfo = '0xd93d3bdba18ebcb3317a57119ea44ed2cf41c2f2';
+
+// Giv - Dai Mainnet LP
+const givDaiPairMainnetAddress = '0xbeba1666c62c65e58770376de332891b09461eeb';
+const givDaiPairUnipoolContractInfo =
+  '0xa4523d703f663615bd41606b46b58deb2f926d98';
+
+// GIV - ETH Balancer
+const givEthBalancerPoolId =
+  '0x7819f1532c49388106f7762328c51ee70edd134c000200000000000000000109';
+const givEthBalancerAddress = '0xc0dbdca66a0636236fabe1b3c16b1bd4c84bb1e1';
+
+// ICHI Angel Vault
+const oneGivPairUnipoolContractInfo =
+  '0xa4b727df6fd608d1835e3440288c73fb28c4ef16';
+
+const defiLlamaGivId = `ethereum:${givTokenMainnetAddress}`;
+
+const tokenPairPoolQuery = gql`
+  {
+    pair(id: "<PLACEHOLDER>") {
+      id
+      reserveUSD
+      volumeUSD
+      reserve0
+      reserve1
+      totalSupply
+      token0 {
+        id
+        symbol
+      }
+      token1 {
+        id
+        symbol
+      }
+    }
+  }
+`;
+
+const balancerPairQuery = gql`
+  {
+    pool(id: "<PLACEHOLDER>") {
+      id
+      tokens {
+        id
+        address
+        balance
+        weight
+        symbol
+      }
+      totalShares
+    }
+  }
+`;
+
+const unipoolContractInfoQuery = gql`
+  {
+    unipool(id: "<PLACEHOLDER>") {
+      id
+      totalSupply
+      rewardRate
+    }
+  }
+`;
+
+const getPrices = async (tokens = []) => {
+  const idsSet = [defiLlamaGivId, ...tokens];
+  let prices = await utils.getData('https://coins.llama.fi/prices', {
+    coins: idsSet,
+  });
+  return prices.coins;
+};
+
+const balancerCalculatePoolApy = async (chainString, entry, contractInfo) => {
+  const givTokenAddress =
+    chainString === 'ethereum' ? givTokenMainnetAddress : givTokenGnosisAddress;
+
+  const weights = [entry.tokens[0].weight, entry.tokens[1].weight].map(
+    toBigNumberJs
+  );
+  const balances = [entry.tokens[0].balance, entry.tokens[1].balance].map(
+    toBigNumberJs
+  );
+
+  if (entry.tokens[0].address.toLowerCase() !== givTokenAddress) {
+    balances.reverse();
+    weights.reverse();
+  }
+
+  const lp = BigNumber(entry.totalShares)
+    .div(BigNumber.sum(...weights).div(weights[0]))
+    .div(balances[0]);
+
+  const totalSupply = BigNumber(contractInfo.totalSupply);
+  const apr = totalSupply.isZero()
+    ? null
+    : BigNumber(contractInfo.rewardRate)
+        .div(totalSupply)
+        .times(secsInOneYear)
+        .times('100')
```

Comment:
I see times '100' in many calculations, what does this represent?
Lets keep numbers in variables for clarity.

Reply:
this is basically to have APR in the 0-100% range - otherwise, the result would be between 0-1 cc @mateodaza

Reply:
Thanks.